### ISSUE TITLE
Research spike on Docusign integration

### ISSUE THREAD

**Issue by shaun-technovation** [Iridescent-CM/technovation-app#4403]
We need a way to know if someone signed/checked all the documents DocuSign and display the results as a part of chapter ambassador onboarding.

- The "Memorandum of Understanding" onboarding step should only be marked as complete if all the documents have been signed

If we can, should we display individual checkboxes for each document in DocuSign and whether or not they have been completed?

**Comment by dboyer** [comment]
We need to have ChAs sign the MOU before they can access information about students in the chapter or set their chapter profile fields. Specific requirements for using Docusign include:

- As part of the MOU, the ChA agrees to several additional documents (Code of Conduct, etc). Those documents are not signed or reviewed in DocuSign. Ideally, we would have them linked on the MOU tab on the ChA dashboard and the ChA would need to click to open each one before they could check the box that they agreed to it (see mockup below).
- We create a Chapter MOU as a template in Docusign. We need to be able to generate an individual doc for each ChA to sign based on that template. 
- It would be ideal to fill in the organization name and individual's name in the document on Docusign based on the fields we send from the platform. If that's not possible, the ChA can fill in those fields.
- This is a single signature document - we will not need to wait on a signature from someone at Technovation after the ChA signs
- After signing, we need to use the Docusign API to update the platform that the doc has been signed and the ChA now has the ability to edit the program profile
- Ideally, the ChA would have access to a copy of their signed MOU on the MOU tab of their onboarding dashboard. They should also see links to the additional documents they agreed to (Code of Conduct, Social Media Policy, etc).

<img width="495" alt="Screen Shot 2024-03-18 at 11 55 42 AM" src="https://github.com/Iridescent-CM/technovation-app/assets/516764/311e457f-b4bf-4dc8-9c38-f5f073d29220">

**Comment by dboyer** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Technovation-605271851956b1000ec2f24d/issues/Iridescent-CM/technovation-app/4403?planning-poker) with Zenhub @shaun-technovation @viviancan

**Comment by shaun-technovation** [comment]
For this spike, I worked on proving out the access token, this work was spun off and completed in #4612. I added notes in this comment: https://github.com/Iridescent-CM/technovation-app/issues/4612#issuecomment-2050305334. I also proved out using the eSignature REST API to send an email to request a signature on a document, this work is being cleaned up and will be complete/functional in #4647. I'll also add notes/findings in that issue.

A few ways/suggestions for future work:
- Adding the "Legal Contact" area to the admin to send an MOU (this will use the new functionality listed above)
- Adding the front-end and back-end functionality to send the chapter document
- A way on our end to track which/when documents were signed
